src/sw.ts

```diff
@@ -476,6 +480,83 @@ async function fetchHandler ({ path, request, event }: FetchHandlerArg): Promise
   }
 }
 
+/**
+ * TODO: better styling
+ * TODO: more error details from @helia/verified-fetch
+ */
+async function errorPageResponse (fetchResponse: Response): Promise<Response> {
+  const responseContentType = fetchResponse.headers.get('Content-Type')
+  let json: Record<string, any> | null = null
+  let text: string | null = null
+  if (responseContentType != null) {
+    if (responseContentType.includes('text/html')) {
+      return fetchResponse
+    } else if (responseContentType.includes('application/json')) {
+      // we may eventually provide error messaging from @helia/verified-fetch
+      json = await fetchResponse.json()
+    } else {
+      text = await fetchResponse.text()
+      json = { error: { message: fetchResponse.statusText, stack: text } }
```

Comment:
nit: for this `text` path, is the plan to fix low-hanging fruit for now and include `err` in message returned in [`return badGatewayResponse(resource, 'Error walking path')`](https://github.com/ipfs/helia-verified-fetch/blob/089635d6cd5b10aefbed013e95637ddb90b166e5/packages/verified-fetch/src/verified-fetch.ts#L257) for now? 

If so, we should not pretend it is stack, but concatenate error message:

```suggestion
      text = await fetchResponse.text()
      json = { error: { message: `${fetchResponse.statusText}: ${text}, stack: null } }
```